### ISSUE TITLE
Deploying and publishing Azure Function (Python) with custom Docker image.

### ISSUE THREAD

**Issue by ktdinh** [Azure/azure-functions-python-worker#331]
I followed [this guide](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-linux-custom-image), to create a custom Docker image hosting an HttpTrigger using Python.

Step 2: When it is time to deploy and publish my Azure Function to Azure, I followed [this section](https://docs.microsoft.com/en-us/azure/azure-functions/functions-create-function-linux-custom-image#create-and-deploy-the-custom-image), and executed 
`az functionapp create --name <app_name> --storage-account  <storage_name>  --resource-group myResourceGroup \
--plan myAppServicePlan --deployment-container-image-name <docker-id>/mydockerimage:v1.0.0`

After step 2 from above, Azure Dashboard showed an **Azure Function App** with empty **Functions** section.  See the attached image, ![image](https://user-images.githubusercontent.com/4603182/53054027-f5136980-3457-11e9-8249-bd65b5f7ae83.png)

I also executed these to set AzureWebJobsDashboard and AzureWebJobsStorage settings.
`$storageConnectionString=$(az storage account show-connection-string \`
`--resource-group myResourceGroup --name <storage_account> \`
`--query connectionString --output tsv)`

`az functionapp config appsettings set --name <function_app> \`
`--resource-group myResourceGroup \`
`--settings AzureWebJobsDashboard=$storageConnectionString \`
`AzureWebJobsStorage=$storageConnectionString`

I would expect that the Azure Function code (HttpTrigger) was already part of the Docker image and it would just show up.  
Is it necessary to run `func azure functionapp publish <app_name>` after step 2 above to actually publish Azure Function code?  If this is the case, it would be helpful to clarify it in the documentation.

Thanks!

**Comment by asavaritayal** [comment]
@ktdinh are you able to successfully run the function locally?

**Comment by ktdinh** [comment]
There was no issue in running the function locally.

**Comment by carterpy** [comment]
I can't see the function either

**Comment by kabirkhan** [comment]
I've also seen this issue but the functions will appear after 5-20min. It's confusing and seems broken only because there's no status message and logging is disabled by default.

**Comment by Hazhzeng** [comment]
The wait is due to your App Service needs to download the Docker image from your registry (e.g. Docker IO). The time may vary due to the size of your Docker image. You may need to manually query the deployment status via the functions list using ARM request:
1. Head to https://resources.azure.com/api/token?plaintext=true to acquire your ARM Bearer token
2. Send a **GET** request to https://management.azure.com/subscriptions/[subscription_uuid]/resourceGroups/[resource_group_name]/providers/Microsoft.Web/sites/[function_app_name]/hostruntime/admin/functions?api-version=2015-08-01
3. In the response, you will see a list of functions inside your function app.

**Comment by ktdinh** [comment]
Thank you @Hazhzeng for getting to the bottom of this issue. That sounds like a reasonable explanation of what's going on in the background. Unfortunately, I no longer have access to the resources to verify this behavior. I will close the issue instead. If the issue persist, other can open a new ticket to track it.
